src/app/events/MessageCreate.ts

```diff
@@ -11,7 +11,7 @@ export default class implements DiscordEvent {
 
 	async execute(message: Message): Promise<any> {
 		try {
-			if(message.author.bot || message.webhookId === null) return;
```

Comment:
The DM message interaction between the bot and the user should only originate from the BanklessDAO discord server. Today there are 5 discords that use this bot. Very quickly this is going to expand to the dozens. I noticed that these variables 

```typescript
// ScoapSquad state initialization
export const scoapEmbedState = {};
export const botConvoState = {};
export const voteRecordState = {};
```

are used to maintain the state of the conversation. Have you taken a look at how the bounty and poap DM conversations take place? Here is a short snippet of how private conversations are held 

```typescript
const dmChannel: DMChannel = await guildMember.createDM();
	const replyOptions: AwaitMessagesOptions = {
		max: 1,
		time: 180000,
		errors: ['time'],
	};
	request.name = (await dmChannel.awaitMessages(replyOptions)).first().content;

	await guildMember.send({
		embeds: [
			{
				title: 'Description',
				description: 'What is this POAP event about? Tell me all of the amazingness about this event that your participants will see!',
			},
		],
	});
	request.description = (await dmChannel.awaitMessages(replyOptions)).first().content;

	await guildMember.send({
		embeds: [
			{
				title: 'Virtual Event',
				description: 'Is this a virtual event? (y/n)',
			},
		],
	});
```

The import thing here is that database storage is not needed to hold the conversation.

Reply:
'The DM message interaction between the bot and the user should only originate from the BanklessDAO discord server.' --> The MessageCreate event is only used for the conversation with the bot within the DM channel and there is a safeguard in place to make sure it only accepts direct responses to the last message sent by the bot (in 'messageSetScoapRoles.ts' 'messageIsValid'). The DM interaction is invoked directly from the command ('command = CreateNewScoapPoll(guildMember, ctx);' in ScoapSquad.ts) so the permission setting should make sure the DM always originates from a permissioned guild.

As of the state: The state variables are used for two things. 1. To be able to retrieve the state of the conversation within the MessageCreate event. Since I can not pass anything to the event I used this method to retrieve the collected user input. Nothing from the bot conversation is written to storage. 2. To keep track of the polls. Once published the scoap polls are open for voting and I used those variables to keep track of the ongoing votes. This is written to storage only for the reason that, if the bot goes down while the poll is still open, we can now retrieve it and restore the poll and voting can continue. Polls are open for a limited time, after the poll completes or times out, data is removed from the db. So the only thing ever written to db is the state of currently active polls. To make it easy to convert from db back to object i just save the whole object in the db. this could be optimized at some point if needed.